cmd/drand-cli/cli.go

```diff
@@ -603,21 +603,28 @@ func resetCmd(c *cli.Context) error {
 	return nil
 }
 
-func askPort() string {
+func askPort(c *cli.Context) string {
 	for {
-		var port string
 		fmt.Fprintf(output, "No valid port given. Please, choose a port number (or ENTER for default port 8080): ")
-		if _, err := fmt.Scanf("%s\n", &port); err != nil {
+
+		reader := bufio.NewReader(c.App.Reader)
+		input, err := reader.ReadString('\n')
+		if err != nil {
 			continue
 		}
+
+		port := strings.TrimSpace(input)
 		if port == "" {
+			fmt.Fprintln(output, "Default port selected")
 			return defaultPort
 		}
-		_, err := strconv.Atoi(port)
-		if len(port) > 2 && len(port) < 5 && err == nil {
-			return port
+
+		_, err = strconv.Atoi(port)
+		if err != nil || len(port) < 4 || len(port) > 6 {
```

Comment:
I would use the result of the `Atoi` method and check if `port > 1000 && port < 65536`